org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/AbstractNotificationPopup.java

```diff
@@ -0,0 +1,567 @@
+    button.addMouseTrackListener(new MouseTrackAdapter() {
+      @Override
+      public void mouseEnter(MouseEvent e) {
+        button.setImage(SonarLintImages.NOTIFICATION_CLOSE_HOVER);
+      }
+
+      @Override
+      public void mouseExit(MouseEvent e) {
+        button.setImage(SonarLintImages.NOTIFICATION_CLOSE);
+      }
+    });
+    button.addMouseListener(new MouseAdapter() {
+
+      @Override
+      public void mouseUp(MouseEvent e) {
+        close();
+        setReturnCode(CANCEL);
+      }
+
+    });
+  }
+
+  protected Color getTitleForeground() {
+    return color.getTitleText();
+  }
+
+  private void initResources() {
+    color = new GradientColors(display, resources);
+  }
+
+  @Override
+  protected void configureShell(Shell newShell) {
+    super.configureShell(newShell);
+
+    shell = newShell;
+    newShell.setBackground(color.getBorder());
+  }
+
+  @Override
+  public void create() {
+    super.create();
+    addRegion(shell);
+  }
+
+  private void addRegion(Shell shell) {
+    Region region = new Region();
+    Point s = shell.getSize();
+
+    /* Add entire Shell */
+    region.add(0, 0, s.x, s.y);
+
+    /* Subtract Top-Left Corner */
+    region.subtract(0, 0, 5, 1);
+    region.subtract(0, 1, 3, 1);
+    region.subtract(0, 2, 2, 1);
+    region.subtract(0, 3, 1, 1);
+    region.subtract(0, 4, 1, 1);
+
+    /* Subtract Top-Right Corner */
+    region.subtract(s.x - 5, 0, 5, 1);
+    region.subtract(s.x - 3, 1, 3, 1);
+    region.subtract(s.x - 2, 2, 2, 1);
+    region.subtract(s.x - 1, 3, 1, 1);
+    region.subtract(s.x - 1, 4, 1, 1);
+
+    /* Subtract Bottom-Left Corner */
+    region.subtract(0, s.y, 5, 1);
+    region.subtract(0, s.y - 1, 3, 1);
+    region.subtract(0, s.y - 2, 2, 1);
+    region.subtract(0, s.y - 3, 1, 1);
+    region.subtract(0, s.y - 4, 1, 1);
+
+    /* Subtract Bottom-Right Corner */
+    region.subtract(s.x - 5, s.y - 0, 5, 1);
+    region.subtract(s.x - 3, s.y - 1, 3, 1);
+    region.subtract(s.x - 2, s.y - 2, 2, 1);
+    region.subtract(s.x - 1, s.y - 3, 1, 1);
+    region.subtract(s.x - 1, s.y - 4, 1, 1);
+
+    /* Dispose old first */
+    if (shell.getRegion() != null) {
+      shell.getRegion().dispose();
+    }
+
+    /* Apply Region */
+    shell.setRegion(region);
+
+    /* Remember to dispose later */
+    lastUsedRegion = region;
+  }
+
+  private boolean isMouseOver(Shell shell) {
+    if (display.isDisposed()) {
+      return false;
+    }
+    return shell.getBounds().contains(display.getCursorLocation());
+  }
+
+  @Override
+  public int open() {
+    if (shell == null || shell.isDisposed()) {
+      shell = null;
+      create();
+    }
+
+    constrainShellSize();
+    shell.setLocation(fixupDisplayBounds(shell.getSize(), shell.getLocation()));
+
+    if (isFadingEnabled()) {
+      shell.setAlpha(0);
+    }
+    shell.setVisible(true);
+    fadeJob = AnimationUtil.fadeIn(shell, new IFadeListener() {
+      public void faded(Shell shell, int alpha) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1161)